addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/fritzaha/handler/FritzAhaBridgeHandler.java

```diff
@@ -0,0 +1,282 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.fritzaha.handler;
+
+import static org.openhab.binding.fritzaha.FritzAhaBindingConstants.*;
+
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.fritzaha.FritzAhaBindingConstants;
+import org.openhab.binding.fritzaha.config.FritzAhaBridgeConfiguration;
+import org.openhab.binding.fritzaha.internal.discovery.FritzBoxDiscoveryService;
+import org.openhab.binding.fritzaha.internal.hardware.FritzahaWebInterface;
+import org.openhab.binding.fritzaha.internal.hardware.callbacks.FritzAhaDiscoveryCallback;
+import org.openhab.binding.fritzaha.internal.hardware.callbacks.FritzAhaUpdateXmlCallback;
+import org.openhab.binding.fritzaha.internal.model.DeviceModel;
+import org.openhab.binding.fritzaha.internal.model.SwitchModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler for a FRITZ!Box device. Handles polling of values from AHA devices.
+ * 
+ * @author Robert Bausdorf
+ * @since 2.0
+ */
+public class FritzAhaBridgeHandler extends BaseBridgeHandler {
+	/**
+	 * Logger
+	 */
+	private static Logger LOG = LoggerFactory
```

Comment:
Yes, see http://stackoverflow.com/questions/8262310/any-reason-to-use-private-instead-of-private-final-static-on-the-logback-logger
Static definition is IOC-unfriendly and that means that it does not work well within OSGi, where the container manages the lifecycle and wants to dispose bundles and their classes when restarting or uninstalling it.